elastic/kibana
# [Entity Analytics] [8.18 Only] Add deprecation warning for the legacy risk score modules

## Summary

> [!NOTE]  
> Example deployment with data loaded (message me for credentials): https://kibana-pr-202775.kb.us-west2.gcp.elastic-cloud.com/app/management/stack/upgrade_assistant/kibana_deprecations


Add a deprecation warning when users have the risk scoring modules/legacy risk engine installed. 

We are removing the legacy risk engine in 9.0.0 which means users data will be there in the index but the UI won’t display it, scores will even continue to be calculated if they for some reason don’t upgrade to the new risk engine. But there will be no dedicated views in kibana.

In 8.18 we are adding an upgrade assistant warning if they have the legacy risk engine installed, this is to guide them to upgrade.

I have tried to use the same language as our documentation by calling it the risk score module.

Upgrade Assistant List view:
<img width="1728" alt="Screenshot 2024-12-12 at 09 40 36" src="https://github.com/user-attachments/assets/90cbfa44-fc29-42e5-afea-c74b6b848881">



Clicking the deprecation:
<img width="1343" alt="Screenshot 2024-12-13 at 10 59 21" src="https://github.com/user-attachments/assets/8299038e-fe26-44d0-8330-7682fcdf29c9" />



### Test Steps

- on a system with the legacy risk module installed
    - I used the [security-documents-generator](https://github.com/elastic/security-documents-generator) `yarn start generate-legacy-risk-score` command.
    - You would have to deploy an 8.11 system to install it through the UI
- navigate the the update assistant in stack management
- notice the deprecation warning is displayed
- upgrade your risk engine by navigating to security > management > entity risk score and clicking start upgrade
- notice the deprecation warning is no longer displayed.
